banyand/measure/part_iter.go

```diff
@@ -207,7 +207,9 @@ func (pi *partIter) readPrimaryBlock(bms []blockMetadata, mr *primaryBlockMetada
 	}
 	bmPtrs := make([]*blockMetadata, 0, len(bms))
 	for _, bm := range bms {
-		bmPtrs = append(bmPtrs, &bm)
+		bmCopy := &blockMetadata{}
```

Comment:
@ButterBright, copying it tends to fix an issue, that is, `bm` is already cached by another pool.